File: voidshare/p2p_encryption/main.py
from p2p.peer import Peer
import threading

def start_receiver():
    receiver = Peer()
    receiver.receive_file("received.txt")

def start_sender():
    sender = Peer()
    sender.send_file("file_to_send.txt", "localhost", 5000)

# Simulating P2P transfer
receiver_thread = threading.Thread(target=start_receiver)
receiver_thread.start()

sender_thread = threading.Thread(target=start_sender)
sender_thread.start()

File: voidshare/p2p_encryption/main.ts
import { Peer } from './p2p/peer';

const sender = new Peer(5001);
const receiver = new Peer(5000);

// Simulate saving public keys
import * as fs from 'fs';
fs.writeFileSync('sender_pub.pem', sender.publicKey);
fs.writeFileSync('receiver_pub.pem', receiver.publicKey);

// Start receiver
receiver.receiveFile('received.txt');

// Simulate file send
setTimeout(() => {
  sender.sendFile('file_to_send.txt', 'localhost', 5000);
}, 2000);
